### PR TITLE
 raftstore: split store module

### PR DESCRIPTION
**Pull request by BusyJay**
This PR divides store module into two parts: one is for overall store
operations, the other is for region related operations. It's very
helpful for threaded raftstore experiment, as it can reduce the conflict
a lot.

### CODE REVIEW

**Comment by BusyJay** [comment]
No new functionality is added, just moving code around and adjusting method visibility. PTAL thanks!

**Comment by BusyJay** [comment]
Finite State Machine. Both `Store` and `Peer` will be refactored to FSM eventually.

**Comment by siddontang** [comment]
@BusyJay 

Maybe it is better to add some comments in the fsm/mod.rs to explain fsm?

File: src/raftstore/store/fsm/store.rs [hunk 1]
```diff
@@ -0,0 +1,1256 @@
+// Copyright 2016 PingCAP, Inc.
```
Comment (overvenus) [inline src/raftstore/store/fsm/store.rs]: 2018

Comment (BusyJay) [reply src/raftstore/store/fsm/store.rs]: Actually the file is moved and `peer.rs` is a new file.

**Pull request merged**